lib/cucumber/core/test/action.rb

```diff
@@ -21,8 +22,16 @@ def skip(*)
 
         def execute(*args)
           @timer.start
-          @block.call(*args)
-          passed
+          invoke_result = @block.call(*args)
+
+          case invoke_result
```

Comment:
minor nit, to improve number of lines this could be written as (Which I think looks a bit clearer)

```
case invoke_result
when PassedInvokeResult; then passed(invoke_result.embeddings)
when FailedInvokeResult; then failed(invoke_result.exception, invoke_result.embeddings)
else passed
end
```

Note the couple of slight differences from strongly/weakly typed (no return / blank parentheses)

Reply:
Not a huge fan of the implicit return (this piece of code actually the first of 7 layers of implicit return), but OK.